src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1014,7 +1094,7 @@ private void Filter_SelectionChanged(object sender, FilterChangedEventArgs e)
         /// </summary>
         private void Refresh()
         {
-            if (_topPanel.Filter != ItemFilter.All)
+            if (_topPanel.Filter != ItemFilter.All || _recommendPackages)
```

Comment:
Can you elaborate why we need this check? 

As is, we'd refresh a lot more often than we should.

Reply:
If a user installs or uninstalls a package, that would change what packages we recommend. So we would want this list to be refreshed if the recommender is being used.

Reply:
That's not the only case where this code path runs. I think the comment is misleading. 

If you look at the invocations of that method there sohuld be like 9 or different code paths calling into it. 

This check was basically "if on install/updates/consolidate" tab, refresh the whole list...but now we are refreshing when on the browse tab too even if nothing has changed.

Reply:
I've removed this for now while I look for a more targeted way to trigger a refresh.